Promote string/array list unknown encoding to ValueError in MBString

Girgias opened the pull request:
So it seems incorrect INI settings don't display the error so I'm not sure how to approach this, any ideas would be welcomed.

nikic commented on ext/mbstring/mbstring.c:
```diff
@@ -420,7 +420,7 @@ php_mb_parse_encoding_list(const char *value, size_t value_length, const mbfl_en
 					n++;
 				} else {
-					php_error_docref(NULL, E_WARNING, "Unknown encoding \"%s\"", p1);
+					zend_argument_value_error(arg_num, "must be a list of valid encoding, \"%s\" given", p1);
```
```suggestion
					zend_argument_value_error(arg_num, "must be a list of valid encodings, \"%s\" given", p1);
```

Though I think it is a bit weird, because `%s` is only one of the encodings in the list. Another possibility would be:

```suggestion
					zend_argument_value_error(arg_num, "contains invalid encoding \"%s\"", p1);
```

nikic commented:
> So it seems incorrect INI settings don't display the error so I'm not sure how to approach this, any ideas would be welcomed.

Is there a test for that?

Maybe check if arg_num==0 and in that case do E_WARNING instead of exception?

Girgias commented:
> 
> 
> > So it seems incorrect INI settings don't display the error so I'm not sure how to approach this, any ideas would be welcomed.
> 
> Is there a test for that?
> 
> Maybe check if arg_num==0 and in that case do E_WARNING instead of exception?

There wasn't so I've added one with all the INI settings having "bad" values.

Also implemented the suggestion but the message is pretty generic as the INI setting cannot be known. I suppose this could be delegated to the INI handler but this could also be good enough.

nikic commented on ext/mbstring/mbstring.c:
```diff
@@ -475,8 +482,13 @@ php_mb_parse_encoding_array(HashTable *target_hash, const mbfl_encoding ***retur
-				php_error_docref(NULL, E_WARNING,
-					"Unknown encoding \"%s\"", ZSTR_VAL(encoding_str));
+				/* Called from an INI setting modification */
+				if (arg_num == 0) {
```
This function cannot be used for ini settings (only strings make sense there), so this shouldn't be needed.

Girgias replied:
Ah indeed will fix and commit
